src/main/scala/za/co/absa/hyperdrive/trigger/persistance/DagInstanceRepository.scala

```diff
@@ -75,20 +75,23 @@ class DagInstanceRepositoryImpl @Inject()(val dbProvider: DatabaseProvider) exte
   }
 
   def getDagsToRun(runningIds: Seq[Long], size: Int, assignedWorkflowIds: Seq[Long])(implicit executionContext: ExecutionContext): Future[Seq[DagInstance]] = {
-    val prefilteredResult = db.run(
-      dagInstanceTable.filter { di =>
-        !di.workflowId.in(
-          dagInstanceTable.filter(_.id.inSet(runningIds)).map(_.workflowId)
-        )
-      }
-        .filter(_.status.inSet(DagInstanceStatuses.nonFinalStatuses))
-        .filter(_.workflowId inSetBind assignedWorkflowIds)
-        .result
-    )
-
-    prefilteredResult.map(di =>
-      di.groupBy(_.workflowId).flatMap(_._2.sortBy(_.id).take(1)).take(size).toSeq
-    )
+    val dagIdsQuery = dagInstanceTable
+      .filter(_.status inSet DagInstanceStatuses.nonFinalStatuses)
+      .filter(_.workflowId inSetBind assignedWorkflowIds)
+      .filterNot(_.id inSet runningIds)
```

Comment:
Is this simplification ok? In the original query, all dags from the same running workflow would be excluded

Reply:
Yes, you are right. Fixed